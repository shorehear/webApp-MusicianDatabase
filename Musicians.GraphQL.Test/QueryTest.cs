using AutoMapper;
using CookieCrumble;
using HotChocolate;
using HotChocolate.Execution;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using Musicians.Database;
using Newtonsoft.Json.Linq;

namespace Musicians.Test
{ 
    public class QueryTest : IClassFixture<TestService>
    {
        private readonly IServiceProvider serviceProvider;

        public QueryTest(TestService testService) 
        {
            serviceProvider = testService.ServiceProvider;
        }

        [Fact]
        public async Task TestGetMusicians()
        {
            using var scope = serviceProvider.CreateScope();
            var contextFactory = scope.ServiceProvider.GetRequiredService<IDbContextFactory<MusiciansDbContext>>();
            using var context = contextFactory.CreateDbContext();

            SeedTestData.Initialize(context);

            var executor = await scope.ServiceProvider.GetRequiredService<IRequestExecutorResolver>()
                .GetRequestExecutorAsync();

            var result = await executor.ExecuteAsync("{ musicians { musicianName } }");

            result.MatchSnapshot();
        }

        [Fact]
        public async Task TestGetMusiciansByCountry()
        {
            using var scope = serviceProvider.CreateScope();
            var contextFactory = scope.ServiceProvider.GetRequiredService<IDbContextFactory<MusiciansDbContext>>();
            using var context = contextFactory.CreateDbContext();

            SeedTestData.Initialize(context);

            var executor = await scope.ServiceProvider.GetRequiredService<IRequestExecutorResolver>()
                .GetRequestExecutorAsync();

            var result = await executor.ExecuteAsync(@"
                {
                    musiciansByCountry(countryName: ""USA"") {
                        musicianName
                        country {
                            countryName
                        }
                    }
                }");

            result.MatchSnapshot();

            var resultData = JObject.Parse(result.ToJson());

            var musicians = resultData["data"]?["musiciansByCountry"];
            Assert.NotNull(musicians);
            Assert.NotEmpty(musicians);

        }
    }
}
