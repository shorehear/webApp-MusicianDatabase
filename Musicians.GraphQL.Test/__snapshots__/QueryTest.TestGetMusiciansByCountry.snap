{
  "data": {
    "musiciansByCountry": [
      {
        "musicianName": "Paul McCartney",
        "country": {
          "countryName": "USA"
        }
      },
      {
        "musicianName": "John Lennon",
        "country": {
          "countryName": "USA"
        }
      }
    ]
  }
}